src/luascript.cpp

```diff
@@ -6197,11 +6197,11 @@ int LuaScriptInterface::luaItemSetCustomAttribute(lua_State* L) {
 	if (isNumber(L, 2)) {
 		int64_t tmp = getNumber<int64_t>(L, 2);
 		key.type = ATTR_INTEGER_TYPE;
-		key.value.integer = tmp;
+		key.value = tmp;
 	} else if (isString(L, 2)) {
 		const std::string& tmp = getString(L, 2);
 		key.type = ATTR_STRING_TYPE;
-		key.value.string = new std::string(tmp);
+		key.value = std::string(tmp);
```

Comment:
The construction of a temporary std::string is unnecessary - you can just call key.value = getString(...).